### ISSUE TITLE
Attachment to be a FAB button to enable Speedial and other type or menues

### ISSUE THREAD

**Issue by SalahAdDin** [flyerhq/flutter_chat_ui#58]
## Is your feature request related to a problem?
Many chat applications offer a speedial/FAB button menu when attaching files. Right now, the `onAttachmentPressed` just offers a way to pass a callback function (used almost for showing dialogs), but not somekind of components for example.

## What solution would you suggest?
It would be better if we have an option to pass a widget, or make `onAttachmentPressed` available to receive widgets.

Some examples: https://fluttergems.dev/floating-action-button/

## Extras
**Screenshots or videos 📸**
If applicable, add screenshots or videos to help explain your feature.

**Code snippets 📝**
If applicable, add code samples to help explain your feature.

```dart
// Your snippet here...
```

## Related issues/PRs
Let us know if this is related to any issue/pull request.

**Comment by demchenkoalex** [comment]
Hi @SalahAdDin

Attachment button is a part of UI, it can't be FAB or anything else. It does accept a widget if you want to configure the visuals (see `attachmentButtonIcon` inside a theme). Inside `onAttachmentPressed` you can present whatever dialog you like, it is outside of the chat package.

How you think we should accept a widget? where we would build it? It can be like a FAB button in the bottom right corner, it might be on the bar where we have it now.

Also where you usually see FAB button we would have a scroll to bottom indicator in the future.

So in short this is not possible. Unless we do a custom widget for a whole bottom part of the chat, but this will not work either because of unability to control text field and other stuff.

Also please remember that there are thousands of chats, and it is impossible to customize everything so it looks like X or Y. In some cases forking a package and changing it to your liking will be the way to go.

Let me know if you have other thoughts.
